### DISCUSSION TITLE
How to run 'up' migrations on only updated server?

### DISCUSSION THREAD

**Discussion by perqin** [sequelize/umzug#163] [Ideas] [answered]
I know that the README mentions it, but I just can't understand it well.

Let's say that my server app V1 requires a database in some schema A. And later V2 changes to schema to B, later V3 changes to schema C.

So, I will define 2 migrations: `A_to_B` and `B_to_C`. Now that I want the V3 version will:

1. Run `A_to_B` and `B_to_C` on schema A (that is, upgrade from V1 to V3).
2. Run `B_to_C` on schema B (that is, upgrade from V2 to V3).
3. Run nothing on schema C or empty database.

If the requirement 3 is excluded, I can just run all pending migrations, but I want it not to run any migrations on empty database. How to achieve that?

**Comment by ephemer** [comment, marked as answer]
I think you'd just have to add the info that these migrations are already complete into your new (fresh) database, or write your migrations such that they ignore an empty db.
